### PR TITLE
Devel

### LINKED ISSUE #2: Refactor to pypi (Build core functions)
- [x] Make the package pip installable and organize with a `setup.py`. 
- [x] descriptive `README.md`
- [ ] submit to pypi

### LINKED ISSUE #5: Plotting functions for visualization
- [ ] Heatmap showing the latent dimensions
- [ ] UMAP or tSNE 
- [ ] Scatter plot of LVs
- [ ] Other plots from the single-cell space?

### LINKED ISSUE #8: Refactor for PyPi (other logistics)
- [x] Add flake8 linting
- [x] Unit tests
- [x] Coverage (fail under 90%)
- [x] .github actions workflow

### LINKED ISSUE #9: Choose a name
Ideas:

1. `PLIERpy`
2. `autoPLIER`
3. `tensorPLIER`

### LINKED ISSUE #14: Experiment: Classification of cell types using the LVs
Need to test how well the LVs from the model are capable of predicting cell types

### LINKED ISSUE #15: Need to make a low-dimension embedding to see if it is interpretable
Need to make a PCA / UMAP representation of the LV space to see if this is actually interpretable.

### PR DESCRIPTION
**Pull request by millerh1**
Should close #8 #10 #9 #2 #7 #5

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,386 +0,0 @@
-""" generate metabolite data in latent variable representation """
```
Comment (Serhiy-Shekhovtsov) [inline main.py:1]: Did you move this file somewhere?

Comment (millerh1) [reply main.py:1]: Yep -- it doesn't belong in the refactored structure. The current PR is really just a placeholder -- it shouldn't be merged until all tests are passing, which means the refactor needs to be finished and modules need to be working right on the `devel` branch

**Comment by millerh1** [comment]
@dmontemayor @Serhiy-Shekhovtsov @mbernste @Julian-Borbeck Please do not accept this PR until it passes. I made the requirements a bit stringent regarding coverage (fail under 90%) since I think that will save us a lot of time in the long run.

File: autoplier/model.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    def fit(self, x_train, callbacks =[], batch_size=50, maxepoch=2000, verbose=2, valfrac=.3):
+
+        # fit the autoencoder model to reconstruct input
+        history = self.model.fit(x_train, x_train, epochs=maxepoch, batch_size=batch_size, verbose=verbose,
+                                 validation_split=valfrac, callbacks=callbacks)
+        return history
+
+    # - - - - - - Build Encoder Model - - - - - -
+    def build_encoder(self):
+
```
Comment (dmontemayor) [inline autoplier/model.py]: The encoder should be compiled at end of the fit function. User should probably not need to build the final_encoder themselves.  Maybe the transform function can gracefully exit if the final encoder hasn't been fit prior?

File: autoplier/model.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        # compile encoder model- with adam opt and use mse as reconstruction error
+        self.final_encoder.compile(optimizer='adam', loss='mse')
+
+    def transform(self, x_predict, index):
+        z_predicted = pd.DataFrame(self.final_encoder.predict(x_predict), index=index)
+
+        return z_predicted
+
+    def fit_transform(self, x_train, index, callbacks=[], batch_size=50, maxepoch=2000, verbose=2, valfrac=.3):
+        # fit the autoencoder model to reconstruct input
```
Comment (dmontemayor) [inline autoplier/model.py:110]: Some code redundancy especially when passing the default arg values. maybe fit_transform should call the defined fit and transform functions?

File: README.md [hunk 4]
```diff
@@ -1,51 +1,72 @@
...
-```
-Then run
-```
-make
-```
-to create a new virtual environment. Note, everytime `requirements.txt` is modified `make` will update and rebuild the virtual environment.
+# Autoplier
+Embed Omics data using an interpretable Tensorflow based model.
+
+![Build Status](https://github.com/dmontemayor/jdrfmetabo/tree/devel/tests/badge.svg)
```
Comment (dmontemayor) [inline README.md:4]: repo name changed

File: README.md [hunk 5]
```diff
@@ -1,51 +1,72 @@
...
+
+## Overview
+This package contains a Tensorflow model built in spirit of the *PLIER* <sup>1</sup> approach to embed Omics data into a latent space representation.
+
+## Installation
+pip install -U git+https://github.com/dmontemayor/jdrfmetabo.git@devel
```
Comment (dmontemayor) [inline README.md:10]: we can promote the devel branch to the main and avoid installing the branch

File: autoplier/model.py (hunk 2 above)
Comment (Julian-Borbeck) [reply autoplier/model.py]: I have changed this on the devel-tests branch already, I solved it by calling the build_encoder function at the end of the fit. A check at the transform function is a great idea, I will add that

File: autoplier/model.py (hunk 3 above)
Comment (Julian-Borbeck) [reply autoplier/model.py:110]: Yes, I agree, I will fix that asap.

**Comment by millerh1** [comment]
@Julian-Borbeck Looks like we still need some test / example data for this

**Pull request merged**